packages/liveblocks-react/src/lib/querying.ts

```diff
@@ -3,27 +3,42 @@ import type {
   InboxNotificationData,
   ThreadData,
 } from "@liveblocks/client";
-import { isNumberOperator, isStartsWithOperator } from "@liveblocks/core";
+import {
+  getSubscriptionKey,
+  isNumberOperator,
+  isStartsWithOperator,
+} from "@liveblocks/core";
 
 import type { InboxNotificationsQuery, ThreadsQuery } from "../types";
+import type { SubscriptionsByKey } from "../umbrella-store";
 
 /**
  * Creates a predicate function that will filter all ThreadData instances that
  * match the given query.
  */
 export function makeThreadsFilter<M extends BaseMetadata>(
-  query: ThreadsQuery<M>
+  query: ThreadsQuery<M>,
+  subscriptions: SubscriptionsByKey | undefined
 ): (thread: ThreadData<M>) => boolean {
   return (thread: ThreadData<M>) =>
-    matchesThreadsQuery(thread, query) && matchesMetadata(thread, query);
+    matchesThreadsQuery(thread, query, subscriptions) &&
+    matchesMetadata(thread, query);
 }
 
 function matchesThreadsQuery(
   thread: ThreadData<BaseMetadata>,
-  q: ThreadsQuery<BaseMetadata>
+  q: ThreadsQuery<BaseMetadata>,
+  subscriptions: SubscriptionsByKey | undefined
 ) {
   // Boolean logic: query.resolved? => q.resolved === t.resolved
-  return q.resolved === undefined || thread.resolved === q.resolved;
+  const subscription = subscriptions?.[getSubscriptionKey("thread", thread.id)];
```

Comment:
`getSubscriptionKey` is cheap but it's still extra work so maybe we could only call it if `subscriptions` exist?

And we can update/move the comment above about `query.resolved`

Reply:
Good point! I've updated it.

I removed the comment about boolean logic, I don't think it's adding anything.